File: hw/suarez/U3/HW16MazeGeneratorASCII/MazeGenerator/src/ec/edu/espe/model/Room.java
package ec.edu.espe.model;

import java.util.EnumMap;

public class Room {
    private final EnumMap<Direction, Wall> walls;
    private boolean visited;
    private final int row;
    private final int col;

    public Room(int row, int col) {
        this.row = row;
        this.col = col;
        this.walls = new EnumMap<>(Direction.class);
        this.visited = false;
        for (Direction dir : Direction.values()) {
            walls.put(dir, new Wall());
        }
    }

    public Wall getWall(Direction direction) {
        return walls.get(direction);
    }

    public void setWall(Direction direction, Wall wall) {
        walls.put(direction, wall);
    }

    public boolean isVisited() {
        return visited;
    }

    public void setVisited(boolean visited) {
        this.visited = visited;
    }

    public int getRow() {
        return row;
    }

    public int getCol() {
        return col;
    }
}

File: hw/yanez/u3/HW16MazeGenerator/model/Room.py
from model.Direction import Direction
from model.Wall import Wall

class Room:
    def __init__(self, row: int, col: int):
        self.row = row
        self.col = col
        self.visited = False
        self.walls = {direction: Wall() for direction in Direction}

    def get_wall(self, direction: Direction) -> Wall:
        return self.walls[direction]

    def set_wall(self, direction: Direction, wall: Wall):
        self.walls[direction] = wall

    def is_visited(self) -> bool:
        return self.visited

    def set_visited(self, visited: bool):
        self.visited = visited

    def get_row(self) -> int:
        return self.row

    def get_col(self) -> int:
        return self.col
